### PR TITLE
CU-86dqyzpz7: BE: Add push state on bets status

### PR DESCRIPTION
**Pull request by lualreye**
With this pr: 

- Handle status error
- Added 'push' status bets
- Added test cases

### CODE REVIEW

File: src/controllers/bets.ts [hunk 1]
```diff
@@ -112,15 +112,19 @@ async function updateBet(req: Request, res: Response) {
...
+    }
 
     await models.bets.findByIdAndUpdate(id, { $set: { status: status } });
 
     res.send({
       message: 'Bet Status Updated'
     });
   } catch (error) {
-    handleHttpError(res, 'Cannot Update Bet Status');
+    handleHttpError(res, 'Cannot Update Bet Status', 400);
```
Comment (robertouski) [inline src/controllers/bets.ts]: In this case that we have more than one possible error, should not be error code: 500?

File: src/tests/integration/bets/betsUpdates.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      .expect(200);
+
+    expect(response.body.message).toBe('Bet Status Updated');
+  });
+
+  it('Should return invalid status', async () => {
+    const response = await request(app)
+      .patch(`/api/bets/${betId}`)
+      .send({ status: 'InvalidStatus' })
+      .expect(400);
```
Comment (robertouski) [inline src/tests/integration/bets/betsUpdates.integration.test.ts:34]: In my last comment if you change the error code from 400 to 500, here we should made a change too.

**Review by robertouski — CHANGES_REQUESTED** [review]

File: src/controllers/bets.ts (hunk 1 above)
Comment (lualreye) [reply src/controllers/bets.ts]: good catch, this should be 500

File: src/tests/integration/bets/betsUpdates.integration.test.ts (hunk 2 above)
Comment (lualreye) [reply src/tests/integration/bets/betsUpdates.integration.test.ts:34]: I will change this

File: src/controllers/bets.ts (hunk 1 above)
Comment (lualreye) [reply src/controllers/bets.ts]: thinking about this, is better to return a generic error from the handler 443, and just remove this code

**Review by robertouski — APPROVED** [review]

**Pull request merged**